UI/MemStickScreen.cpp

```diff
@@ -25,23 +25,83 @@
 #include "Common/UI/ViewGroup.h"
 
 #include "Common/StringUtils.h"
-#include "Common/File/FileUtil.h"
-#include "Common/File/Path.h"
 #include "Common/System/System.h"
 #include "Common/System/NativeApp.h"
-#include "Common/Data/Text/I18n.h"
 #include "Common/System/Display.h"
+#include "Common/Data/Text/I18n.h"
+
+#include "Common/File/AndroidStorage.h"
+#include "Common/File/FileUtil.h"
+#include "Common/File/Path.h"
+#include "Common/File/DiskFree.h"
 
 #include "Core/Util/GameManager.h"
 #include "Core/System.h"
+#include "Core/Config.h"
 
 #include "UI/MemStickScreen.h"
 #include "UI/MainScreen.h"
 #include "UI/MiscScreens.h"
 
-#include "Core/Config.h"
+static bool FolderSeemsToBeUsed(Path newMemstickFolder) {
+	// Inspect the potential new folder.
+	if (File::Exists(newMemstickFolder / "PSP") || File::Exists(newMemstickFolder / "SYSTEM")) {
+		// Does seem likely. We could add more critera like checking for actual savegames or something.
+		return true;
+	} else {
+		return false;
+	}
+}
+
+static bool SwitchMemstickFolderTo(Path newMemstickFolder) {
+	Path testWriteFile = newMemstickFolder / ".write_verify_file";
+
+	// Doesn't already exist, create.
+	// Should this ever happen?
+	if (newMemstickFolder.Type() == PathType::NATIVE) {
+		if (!File::Exists(newMemstickFolder)) {
+			File::CreateFullPath(newMemstickFolder);
+		}
+		if (!File::WriteDataToFile(true, "1", 1, testWriteFile)) {
+			// settingInfo_->Show(sy->T("ChangingMemstickPathInvalid", "That path couldn't be used to save Memory Stick files."), nullptr);
+			// TODO: Display an error!
+			return UI::EVENT_DONE;
+		}
+		File::Delete(testWriteFile);
+	} else {
+		// TODO: Do the same but with scoped storage? Not really necessary, right? If it came from a browse
+		// for folder, we can assume it exists and is writable, barring wacky race conditions like the user
+		// being connected by USB and deleting it.
```

Comment:
I wonder in what exciting ways we would crash if they did that, then started PPSSPP up again later without the USB device connected...

-[Unknown]

Reply:
We would fail the check in NativeApp.cpp "Couldn't read directory '%s' specified by memstick_dir.txt", and it would ask the user for a new directory.